llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -1067,6 +1080,7 @@ VPlan *VPlan::duplicate() {
   }
   Old2NewVPValues[&VectorTripCount] = &NewPlan->VectorTripCount;
   Old2NewVPValues[&VFxUF] = &NewPlan->VFxUF;
+  Old2NewVPValues[&VF] = &NewPlan->VF;
```

Comment:
I must be honest it's not obvious to me why we need a specialised version for the UF=1 case. Why can't we use `VFxUF` always given VFx1 is a valid use case? It seems to add a bit of extra complexity. I understand in a loop we may want to calculate both a runtime VF (for a second part or something) and a runtime VF x UF (for a canonical induction variable, etc), but if we've filled out `Old2NewVPValues[&VFxUF]` with both VFx1 and VFxUF we can just query accordingly, right?

Reply:
> ut if we've filled out Old2NewVPValues[&VFxUF] with both VFx1 and VFxUF we can just query accordingly, right?

I am not sure what filling them out accordingly means here, perhaps using a pair instead of 2 separate fields? 

There are different places that need both VF and VFxUF and to serve them we would need different values I think, but I might be missing something from your suggestion?